nccl_overlap/jacobi.cpp

```diff
@@ -166,19 +169,25 @@ int main(int argc, char* argv[]) {
     const int ny = get_argval<int>(argv, argv + argc, "-ny", 16384);
     const bool csv = get_arg(argv, argv + argc, "-csv");
 
-    int local_rank = -1;
-    {
+    if(num_devices > 1) {
```

Comment:
Can you add an error check that `local_rank < num_devices`? NCCL does not allow to oversubscribe a GPU with MPS and it would be great to add this missing check with this PR.